src/mnode/mnode-pastel.cpp

```diff
@@ -1174,13 +1174,71 @@ CArtTradeTicket CArtTradeTicket::Create(std::string _sellTnxId, std::string _buy
     ticket.price = sellTicket->askedPrice;
 
     ticket.GenerateTimestamp();
-    
+
+    // In case it is nested it means that we have the artTnxId of the sell ticket
+    // available within the trade tickets.
+    // [0]: original registration ticket's txid
+    // [1]: copy number for a given art
+    std::vector<std::string> artRegTicket_TxId_Serial = CArtTradeTicket::GetArtRegTxIDAndSerialIfResoldNft(sellTicket->artTnxId);
+    if(artRegTicket_TxId_Serial[0].compare("") == 0)
+    {  
+      auto artTicket = ticket.FindArtRegTicket();
+      if (!artTicket)
+      {
+        throw std::runtime_error("Art Reg ticket not found");
+      }
+
+      //Original TxId
+      ticket.SetArtRegTicketTxid(artTicket->GetTxId());
+      //Copy nr.
+      ticket.SetCopySerialNr(std::to_string(sellTicket->copyNumber));
+    }
+    else
+    {
+      //This is the re-sold case
+      ticket.SetArtRegTicketTxid(artRegTicket_TxId_Serial[0]);
+      ticket.SetCopySerialNr(artRegTicket_TxId_Serial[1]);
+    }
     std::string strTicket = ticket.ToStr();
     ticket.signature = CPastelID::Sign(reinterpret_cast<const unsigned char*>(strTicket.c_str()), strTicket.size(), ticket.pastelID, strKeyPass);
     
     return ticket;
 }
 
+std::vector<std::string> CArtTradeTicket::GetArtRegTxIDAndSerialIfResoldNft(const std::string& _txid)
+{
+
+    std::vector<std::string> vRetVal = {"",""};
+
+    try
+    {
+      //Possible conversion to trade ticket - if any
+      auto pNestedTicket = CPastelTicketProcessor::GetTicket(_txid, TicketID::Trade);
+      if(pNestedTicket != nullptr)
+      {
+        auto tradeTicket = dynamic_cast<const CArtTradeTicket*>(pNestedTicket.get());
+        if (!tradeTicket)
```

Comment:
you can just write 
if (tradeTicket) {
        vRetVal[0] = tradeTicket->GetArtRegTicketTxid();
        vRetVal[1] = tradeTicket->GetCopySerialNr();
}

instead of 
       if (!tradeTicket)
       {
          return vRetVal;
        }
        vRetVal[0] = tradeTicket->GetArtRegTicketTxid();
        vRetVal[1] = tradeTicket->GetCopySerialNr();
      }
      else
      {
        return vRetVal;
      }

Reply:
Done.